twake/frontend/src/app/scenes/Client/ChannelsBar/Modals/ChannelMembersList.tsx

```diff
@@ -1,50 +1,70 @@
-import React, { FC } from 'react';
+import React, { FC, useState } from 'react';
+import { Button, Col, Row, Typography, Input } from 'antd';
+import { Search } from 'react-feather';
+
+import { ChannelMemberResource, ChannelResource } from 'app/models/Channel';
+import { UserType } from 'app/models/User';
+
+import Strings from 'services/utils/strings.js';
 import Languages from 'services/languages/languages.js';
 import UsersService from 'services/user/user.js';
-import { UserType } from 'app/models/User';
-import Icon from 'components/Icon/Icon';
-import ObjectModal from 'components/ObjectModal/ObjectModal';
-import { Avatar, Button, Col, Row, Typography, Input } from 'antd';
-import { ChannelMemberResource, ChannelResource } from 'app/models/Channel';
 import Collections from 'services/CollectionsReact/Collections';
-import DepreciatedCollections from 'app/services/Depreciated/Collections/Collections.js';
 import ModalManager from 'services/Modal/ModalManager';
+
+import MemberChannelRow from 'scenes/Client/ChannelsBar/Parts/Header/MemberChannelRow.tsx';
 import ChannelMembersEditor from 'scenes/Client/ChannelsBar/Modals/ChannelMembersEditor';
-import Menu from 'components/Menus/Menu.js';
-import { Search } from 'react-feather';
+
+import ObjectModal from 'components/ObjectModal/ObjectModal';
+import { useUsersListener } from 'app/components/Member/UserParts';
 
 type Props = {
   closable?: boolean;
   channel: ChannelResource;
 };
 
-const { Text, Link } = Typography;
+const { Link } = Typography;
+const defaultLimit = 100;
 
 const ChannelMembersList: FC<Props> = props => {
+  const channelMembersUid: string[] = [];
   const { company_id, workspace_id, id } = props.channel.data;
 
+  const [search, setSearch] = useState('');
+  const [limit, setLimit] = useState(defaultLimit);
+  const [searchedUsers, setSearchedUsers] = useState<string[]>([]);
+
   const collectionPath: string = `/channels/v1/companies/${company_id}/workspaces/${workspace_id}/channels/${id}/members/`;
   const channelMembersCollection = Collections.get(collectionPath, ChannelMemberResource);
+  const channelMembers = channelMembersCollection.useWatcher({}, { limit: limit });
+
+  channelMembers.map(member => {
```

Comment:
This is not a `map` but a `forEach`. Keep map when you need to return a value from another one.
If `channelMembersUid` is only generated from channelMembers and is reactive, then you can use a map but like 

```js
const channelMembersUid = channelMembers.map(member => member.data.user_id)
```

Reply:
Done.